components/node-labeler/cmd/run.go

```diff
@@ -110,6 +113,34 @@ var runCmd = &cobra.Command{
 			log.WithError(err).Fatal("unable to bind controller watch event handler")
 		}
 
+		if err := mgr.GetFieldIndexer().IndexField(context.Background(), &workspacev1.Workspace{}, "status.runtime.nodeName", func(o client.Object) []string {
```

Comment:
Because we index this `status.runtime.nodename` field, we can filter for workspaces on just one node instead of always having to query them all. Sweet!